specification.md

```diff
@@ -1,15 +1,279 @@
+
+* it is signed by the functionary key listed in the related step of the 
+  in-toto layout
+
+#### Detailed format
+
+TODO...
+
+
+
 ## Workflows
 
+There are two main workflows for SBOMit documents and SITs: generation and 
+verification.
+
 ### Generation
 
+#### SBOMit document
+To generate a SBOMit document, one must generate the consituent parts.  For the
+in-toto layouts, sub-layouts, attestations, and links, this process is 
+described in the in-toto project's documentation.  Note however, that for the 
+purposes of SBOMit, the layout is required to include certain information 
+so that this information may be used to populate the SIT.  
+
+Generating the mutator by hand may done by starting with a null mutator and 
+then using tooling to generate a SIT in the correct format.  The SIT file may 
+be modified and the JSON diff may be computed.
+
+Similarly, if a SBOMit document is modified, one may elect to add an addendum
+to modify it instead of generating a fresh document.  This is easily done by
+generating the new SBOMit document and then creating a JSON diff.
+
+#### SIT 
+
+The SIT is generated by first deriving common information from in-toto and
+placing it in an intermediate format called X.  This format is effectively
+a SBOM itself which matches the NTIA format....
+
+Then the tool specified in the layout may be used to derive the SIT in the
+appropriate end format.  Note also, that multiple SITs may be generated from 
+the same SBOMit document.  So, for example, one SBOMit document could generate
+both a SPDX SIT and a CycloneDX SIT.
+
+An important part of generating the SIT is applying the mutator to the output
+of the tool used to derive the SIT.  This adds or modifies information in the 
+resulting SIT as instructed.  This works as though applying a JSON patch to 
+the SIT document.
+
+It is recommended that the SIT generation tool provides a warning if the
+mutator modifies (e.g., replaces or removes) portions of the SIT derived from
+the in-toto information.  This is becuase this will be flagged as a security
+risk when verification of the SBOMit document is performed.
+
 ### Verification
+
+#### SBOMit document
+
+SBOMit document verification proceeds by the following steps:
+
+* verifying the in-toto layout over the provided in-toto metadata.  
+* checking each SIT to ensure they are signed by the correct SIT generator
+* generate each SIT using the relevant tooling to examine the mutator 
+  behavior.  Any parts of the SIT that the mutator has chosen to overwrite or 
+  remove represent a substantial security risk.  These are specifically 
+  flagged during validation.  The user must specify a flag like 
+  --insecure-allow-in-toto-override or respond to a similarly scary alert to 
+  install the software.
+
+If this matches the generated SITs, then the SBOMit document verification
+has succeeded.
+
+#### SIT 
+
+A SIT may be verified in two ways.  
```

Comment:
who verifies an SIT vs an SBOMit document? Should the SIT verifier get some assurance that someone checked the SBOMit document?

Reply:
I don't know if this is possible to do.  A SIT is an SBOM (CycloneDX, SPDX, etc.) and I don't think we can add custom verification (as I understand it).

Reply:
It would be nice if we could at least add an attestation that someone verified the full SBOMit document (maybe as a wrapper around the SBOM verification). Otherwise the end user is only getting a signed SBOM

Reply:
I'm fine but if they have to add functionality to their SBOM verification, then it seems like they could just do the SBOMit document verification.  Or do you think that people will want to just check the attestation and then say it is okay?  

I see checking an attestation as roughly equivalent to signing the SBOM with the attestation key.